src/Parser/Warrior/Arms/Modules/Talents/DefensiveStance.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS/TALENTS/WARRIOR';
+import SpellIcon from 'common/SpellIcon';
+import { formatThousands, formatNumber } from 'common/format';
+import Analyzer from 'Parser/Core/Analyzer';
+import StatisticBox, { STATISTIC_ORDER } from 'Interface/Others/StatisticBox';
+
+class DefensiveStance extends Analyzer {
+  perSecond(amount) {
+    return amount / this.owner.fightDuration * 1000;
+  }
+  totalDamageMitigated = 0;
+
+  get drps() {
+    return this.perSecond(this.totalDamageMitigated);
+  }
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.DEFENSIVE_STANCE_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener('damage', this.handleDamageTaken, {
+      toPlayer: true,
+    });
+  }
+
+  handleDamageTaken(event) {
+    if(this.selectedCombatant.hasBuff(SPELLS.DEFENSIVE_STANCE_TALENT.id)) {
+      this.totalDamageMitigated += (event.mitigated || 0);
```

Comment:
Are you sure about adding all the mitigated damage during def stance? I would assume that this stat shows me how much def stance did for me, not how much damage I mitigated during it. The mitigated-prop from WCL includes values from all sources (like externals, avoidance, armor, etc).

You could use something like this (just ripped that from an old version of boneshield, but you should get the idea on how we calculate the DR for just those spells):
```
const preMitigatedBoneShield = (event.amount + event.absorbed) / (1 - BONE_SHIELD_DR);
this.boneShieldMitigated += preMitigatedBoneShield * BONE_SHIELD_DR;
```